File: src/RayTracer.Engine.Lib/Ray.cpp
#include "Ray.h"

using namespace RayTracer;

Ray::Ray()
	: Position(0.0f, 0.0f, 0.0f), Direction(0.0f, 0.0f, 0.0f)
{

}

Ray::Ray(const Vector3& position, const Vector3& direction)
	: Position(position), Direction(direction)
{

}

File: src/yart-rust/yart-engine/src/geometries/ray.rs
use crate::math::{vector::Vector, vector3::Vector3};

#[derive(Debug)]
pub struct Ray {
    position: Vector3,
    direction: Vector3,
    inverse_direction: Vector3,
}

impl Ray {
    pub fn new(position: &Vector3, direction: &Vector3) -> Self {
        Self {
            position: position.clone(),
            direction: direction.clone(),
            inverse_direction: Vector3::reciprical(direction),
        }
    }

    // TODO: Better nane?
    pub fn new2(position: &Vector3, direction: &Vector3, inverse_direction: &Vector3) -> Self {
        Self {
            position: position.clone(),
            direction: direction.clone(),
            inverse_direction: inverse_direction.clone(),
        }
    }

    pub fn position(&self) -> &Vector3 {
        &self.position
    }

    pub fn direction(&self) -> &Vector3 {
        &self.direction
    }

    pub fn inverse_direction(&self) -> &Vector3 {
        &self.inverse_direction
    }
}
